### PR TITLE
Plugin architecture including AWS example

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: docs/PLUGIN.md [hunk 1]
```diff
@@ -1,19 +1,195 @@
...
+          Write{Write}
+          ReadOne{Read /id}
+        end
+    end
+
+    subgraph Amazon Web Services
+      Aurora[(Aurora)]
+      OpenSearch[(OpenSearch)]
+
+      CDC[Kinesis Data Stream] --> Aurora
```
Comment (bradbanister) [inline docs/PLUGIN.md:62]: I'd switch the arrow to point from Aurora to Kinesis

File: docs/PLUGIN.md [hunk 2]
```diff
@@ -1,19 +1,195 @@
...
+> necessary to support arbitrary configuration information. The plugin architecture
+> might require an initialization step that accepts a configuration provider
+> that the plugin can query to retrieve needed information.
+
+### Frontend Architecture
+
+In order to support cloud function implementations, it will be useful to treat
+the ASP.NET frontend as one of many possible frontend implementations. To do so
+might require separation of the console application from the frontend
+implementation.
```
Comment (bradbanister) [inline docs/PLUGIN.md:160]: I think we are most of the way there. A few configuration-y helpers are still in the ASP.NET Core project that could be moved into a shared DMS frontend project, and some small stuff like XsdMetadataEndpoint need to move into DMS core.

File: docs/PLUGIN.md [hunk 3]
```diff
@@ -1,19 +1,195 @@
...
+
+Each of the big-three cloud providers' serverless function technologies have
+.NET libraries that can extend ASP.NET applications. For AWS and Azure, the
```
Comment (bradbanister) [inline docs/PLUGIN.md:163]: Like I say, we are very close to being able to cloud deploy without being a heavy-weight ASP.NET Core app. I'm confident we'll be able to do things just like Meadowlark, with a pure AWS Lambda frontend.

File: docs/PLUGIN.md (hunk 1 above)
Comment (stephenfuqua) [reply docs/PLUGIN.md:62]: To be honest I'm not sure if it is a push or a pull. I can switch it.

File: docs/PLUGIN.md (hunk 3 above)
Comment (stephenfuqua) [reply docs/PLUGIN.md:163]: I agree, I'm confident that it won't take much to get there.

File: docs/PLUGIN.md (hunk 1 above)
Comment (stephenfuqua) [reply docs/PLUGIN.md:62]:
```suggestion
      Aurora --> CDC[Kinesis Data Stream]
```

**Review by bradbanister — APPROVED** [review]

**Pull request merged**